### PR TITLE
Field types

### PR DESCRIPTION
**Pull request by clarkwinkelmann**

### CODE REVIEW

**Comment by clarkwinkelmann** [comment]
This PR introduces a breaking change: because comas are used to separate values in the `in` validation rule, validation rules are now separated by pipes (`|`) instead of the previously used coma (`,`).

Should a migration take care of existing fields and replace `,` with `|` ? Or should another method be used to migrate these ?

**Comment by luceos** [comment]
I'd recommend a migration here.

**Comment by clarkwinkelmann** [comment]
Note to self: add reminder comas can't be used in the dropdown option labels

**Comment by clarkwinkelmann** [comment]
@rodymolenaar are you interested in giving this feature a spin ? It has been tested on one of my clients test install only until now :wink:

File: js/forum/src/mutateUserBio.js [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
+import TypeFactory from 'flagrow/masquerade/types/TypeFactory';
 
 export default function () {
     override(UserBio.prototype, 'view', function (view) {
         // Load the old user bio.
         let original = app.forum.attribute('masquerade.disable-user-bio') ? null : view();
         let answers = app.forum.attribute('canViewMasquerade') ? this.props.user.bioFields() || [] : [];
 
-        return m('div', {className: 'Masquerade-Bio'}, [
+        return m('.Masquerade-Bio', [
```
Comment (luceos) [inline js/forum/src/mutateUserBio.js]: What happened with the element here?

**Comment by luceos** [comment]
Looking good, love how this extension is growing in size as well.

File: js/forum/src/mutateUserBio.js (hunk 1 above)
Comment (clarkwinkelmann) [reply js/forum/src/mutateUserBio.js]: I used the shortcut notation to create a `div` with class `Masquerade-Bio`, I think this is a lot cleaner visually.

But that's a format preference we could discuss :yum:

Comment (luceos) [reply js/forum/src/mutateUserBio.js]: I wasn't aware about that. Good to know. Please leave as is.

**Comment by luceos** [comment]
@clarkwinkelmann is this ready to be merged?

**Comment by clarkwinkelmann** [comment]
This probably need further testing... I discovered a few issues (fixed with the latest commits) while browsing the test server that was using it. Maybe we should call for testers on Discord

**Pull request merged**